File: ios/PouleParty/Models/RTDBCoding.swift
//
//  RTDBCoding.swift
//  PouleParty
//

import Foundation

/// Coerces a Realtime Database numeric leaf into a `Double`. RTDB hands numbers
/// back as `Double`, `Int`, or `NSNumber` depending on the value, so position
/// decoding (PP-102) normalizes through this helper.
func rtdbDouble(_ value: Any?) -> Double? {
    switch value {
    case let d as Double: return d
    case let i as Int: return Double(i)
    case let n as NSNumber: return n.doubleValue
    default: return nil
    }
}

File: android/app/src/main/java/dev/rahier/pouleparty/model/RtdbCoding.kt
package dev.rahier.pouleparty.model

/**
 * Coerces a Realtime Database numeric leaf into a [Double] / [Long]. RTDB hands
 * numbers back as `Long` or `Double` depending on the value, so position
 * decoding (PP-102) normalizes through these helpers.
 */
fun rtdbDouble(value: Any?): Double? = when (value) {
    is Double -> value
    is Long -> value.toDouble()
    is Int -> value.toDouble()
    is Float -> value.toDouble()
    else -> null
}

fun rtdbLong(value: Any?): Long? = when (value) {
    is Long -> value
    is Int -> value.toLong()
    is Double -> value.toLong()
    else -> null
}
